src/main/java/org/opensearch/security/securityconf/ConfigModelV7.java

```diff
@@ -348,101 +349,71 @@ public SecurityRoles filter(Set<String> keep) {
-                        Set<String> concreteIndices = ip.getResolvedIndexPattern(user, resolver, cs);
-
-                        Set<String> currentMaskedFields = maskedFieldsMap.get(indexPattern);
-                        if (currentMaskedFields != null) {
-                            currentMaskedFields.addAll(maskedFields);
-                        } else {
-                            maskedFieldsMap.put(indexPattern, new HashSet<>(maskedFields));
-                        }
-
-                        for (String concreteIndex : concreteIndices) {
-                            currentMaskedFields = maskedFieldsMap.get(concreteIndex);
-                            if (currentMaskedFields != null) {
-                                currentMaskedFields.addAll(maskedFields);
-                            } else {
-                                maskedFieldsMap.put(concreteIndex, new HashSet<>(maskedFields));
-                            }
-                        }
-                    }
-                }
-            }
-            return maskedFieldsMap;
-        }
 
         @Override
-        public Tuple<Map<String, Set<String>>, Map<String, Set<String>>> getDlsFls(User user, IndexNameExpressionResolver resolver,
-                ClusterService cs) {
+        public EvaluatedDlsFlsConfig getDlsFls(User user, IndexNameExpressionResolver resolver, ClusterService cs,
+                NamedXContentRegistry namedXContentRegistry) {
 
-            final Map<String, Set<String>> dlsQueries = new HashMap<String, Set<String>>();
-            final Map<String, Set<String>> flsFields = new HashMap<String, Set<String>>();
 
-            for (SecurityRole sr : roles) {
-                for (IndexPattern ip : sr.getIpatterns()) {
-                    final Set<String> fls = ip.getFls();
-                    final String dls = ip.getDlsQuery(user);
-                    final String indexPattern = ip.getUnresolvedIndexPattern(user);
-                    Set<String> concreteIndices = new HashSet<>();
-
-                    if ((dls != null && dls.length() > 0) || (fls != null && fls.size() > 0)) {
-                        concreteIndices = ip.getResolvedIndexPattern(user, resolver, cs);
-                    }
+            if (!containsDlsFlsConfig()) {
+                if(log.isDebugEnabled()) {
+                    log.debug("No fls or dls found for {} in {} sg roles", user, roles.size());
+                }
 
-                    if (dls != null && dls.length() > 0) {
+                return EvaluatedDlsFlsConfig.EMPTY;
+            }
+        	
+            Map<String, Set<String>> dlsQueriesByIndex = new HashMap<String, Set<String>>();            
+            Map<String, Set<String>> flsFields = new HashMap<String, Set<String>>();
+            Map<String, Set<String>> maskedFieldsMap = new HashMap<String, Set<String>>();
+                        
+            for (SecurityRole sgr : roles) {
```

Comment:
Please do not have `sg` in any names if unnecessary.

Reply:
Fixed this.